lab_1/main.py

```diff
@@ -4,80 +4,107 @@
 """
 
 
-def tokenize(text: str) -> list or None:
-    """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation and other symbols from words
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
-    pass
+def tokenize(text: str):
+    if isinstance(text, str):
+        text = text.lower()
+        text = text.replace("'", ' ')
+        n_text = ''
+        for i in text:
+            if i.isalpha() or i == ' ' or i == '\n':
+                n_text += i
+            else:
+                text.replace(i, ' ')
+        t_tokens = n_text.split()
+        return t_tokens
+    return None
 
 
 def remove_stop_words(tokens: list, stop_words: list) -> list or None:
-    """
-    Removes stop words
-    :param tokens: a list of tokens
-    :param stop_words: a list of stop words
-    :return: a list of tokens without stop words
-    """
-    pass
+    if isinstance(tokens, list) and isinstance(stop_words, list):
+        no_stop_words = []
+        for token in tokens:
+            if token not in stop_words:
+                no_stop_words.append(token)
+        return no_stop_words
+    return None
 
 
-def calculate_frequencies(tokens: list) -> dict or None:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens
-    :return: a dictionary with frequencies
-    """
-    pass
 
+def calculate_frequencies(tokens: list) -> dict or None:
+    if isinstance(tokens, list):
+        f_dict = {}
+        for token in tokens:
+            if type(token) != str:
+                return None
+            if token not in f_dict:
+                f_dict[token] = 1
+            else:
+                f_dict[token] += 1
+        return f_dict
+    return None
 
-def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
-    """
-    Returns the most common words
-    :param freq_dict: a dictionary with frequencies
-    :param top_n: a number of the most common words
-    :return: a list of the most common words
-    """
-    pass
 
 
-def create_language_profile(language: str, text: str, stop_words: list) -> dict or None:
-    """
-    Creates a language profile
-    :param language: a language
-    :param text: a text
-    :param stop_words: a list of stop words
-    :return: a dictionary with three keys – name, freq, n_words
-    """
-    pass
+def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
+    if isinstance(freq_dict, dict) and isinstance(top_n, int):
```

Comment:
There should be a check for the input arguments at the top of the function - after the documentation string. Remove nesting here and extract the check into a separate `if` block earlier.